### PR TITLE
Check argument list is complete [1-25]

### LINKED ISSUE #11: Error expected if argument list is incomplete [1-25]
For non-defined but required inputs the rendering did not throw an error.
Consider the example

```
import sys
import yaml
from dewret.tasks import task, run
from dewret.renderers.cwl import render

@task()
def increment(num: int) -> int:
    return num + 1

result = increment()
workflow = run(result)
cwl = render(workflow)
yaml.dump(cwl, sys.stdout, indent=2)
```
but instead create (to my understanding incorrect) an output:
```
class: Workflow
cwlVersion: 1.2
inputs: {}
outputs:
  out:
    label: out
    outputSource: increment-ecfc9b657ab3a0f2bc96ee9dce5e98e3/out
    type: int
steps:
  increment-ecfc9b657ab3a0f2bc96ee9dce5e98e3:
    in: {}
    out:
    - out
    run: increment
````

### PR DESCRIPTION
**Pull request by philtweir**
Closes #11 - provides basic Python-valid checking of task signatures in calls. Both directly, and within a nested task.

This also adds a series of minor CI fixes that arose during implementation, to ensure tests and static checking is thorough and passes.

### CODE REVIEW

File: src/dewret/tasks.py [hunk 1]
```diff
@@ -260,15 +269,15 @@ def _fn(*args: Any, __workflow__: Workflow | None = None, **kwargs: Param.kwargs
             original_kwargs = dict(kwargs)
             for var, value in inspect.getclosurevars(fn).globals.items():
                 if var in kwargs:
-                    raise RuntimeError("Captured parameter (global variable in task) shadows an argument")
+                    raise TypeError("Captured parameter (global variable in task) shadows an argument")
```
Comment (elleryames) [inline src/dewret/tasks.py:272]: I don't think we have a test for this do we? It would help me understand what we are guarding against better.

Comment (philtweir) [reply src/dewret/tasks.py:272]: Yes, In the end I decjded it was redundant (trying to use `global` to pull in a global variable with the same name as a parameter is a syntax error, besides being an edge case) and I think it was removed in the next PR...

**Review by elleryames — APPROVED** [review]
Aside from a potentially missing test, this seems reasonable.

File: src/dewret/tasks.py (hunk 1 above)
Comment (elleryames) [reply src/dewret/tasks.py:272]: ah, okay, that makes sense.

Comment (philtweir) [reply src/dewret/tasks.py:272]: Confirmed, gone in #15 as part of the error message refactor.

**Comment by philtweir** [comment]
> Aside from a potentially missing test, this seems reasonable.

Given that call is going away in a stacked PR anyway, I'll take that as good to go?
